### PR TITLE
mqttcli: new port (version 0.2.6)

### PR DESCRIPTION
**Pull request by subpop**
#### Description

Add a new port for `mqttcli`, a simple commandline MQTT client.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix
- [x] new port

###### Tested on

macOS 14.4.1 23E224 arm64
Xcode 15.3 15E204a

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [x] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?
- [x] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: net/mqttcli/Portfile [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+post-build {
+    system "a2x --doctype manpage --format manpage --destination-dir ${worksrcpath} ${worksrcpath}/doc/sub.adoc"
+    system "a2x --doctype manpage --format manpage --destination-dir ${worksrcpath} ${worksrcpath}/doc/pub.adoc"
+}
+
+destroot {
+    xinstall -m 0755 ${worksrcpath}/pub ${destroot}${prefix}/bin/
+    xinstall -m 0755 ${worksrcpath}/sub ${destroot}${prefix}/bin/
+    xinstall -m 0644 ${worksrcpath}/pub.1 ${destroot}${prefix}/share/man/man1/
+    xinstall -m 0644 ${worksrcpath}/sub.1 ${destroot}${prefix}/share/man/man1/
```
Comment (ryandesign) [inline net/mqttcli/Portfile:75]: Can be shortened to:
```suggestion
    xinstall -W ${worksrcpath} pub sub ${destroot}${prefix}/bin/
    xinstall -W ${worksrcpath} -m 0644 pub.1 sub.1 ${destroot}${prefix}/share/man/man1/
```

Comment (subpop) [reply net/mqttcli/Portfile]: Nice! Thank you for the tip.

File: net/mqttcli/Portfile [hunk 2]
```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem           1.0
+PortGroup            golang 1.0
+
+go.setup             github.com/subpop/mqttcli 0.2.6 v
```
Comment (herbygillot) [inline net/mqttcli/Portfile:6]: We usually also add a `revision` field.

`revision` is used to indicate that a port has changed/upated in a situation where its current version isn't changing.

Comment (subpop) [reply net/mqttcli/Portfile:6]: Oh yes. I forgot about that. Thanks for the reminder!

**Review by herbygillot — APPROVED** [review]

**Comment by herbygillot** [comment]
Thank you for this submission, @subpop.

**Pull request merged**